pkg/blob/controllerserver.go

```diff
@@ -667,6 +683,103 @@ func (d *Driver) DeleteBlobContainer(ctx context.Context, subsID, resourceGroupN
 	})
 }
 
+// CopyBlobContainer copies a blob container in the same storage account
+func (d *Driver) CopyBlobContainer(ctx context.Context, subsID, resourceGroupName, accountName, accountKey, srcContainerName, dstContainerName string, secrets map[string]string) error {
+	if srcContainerName == "" {
+		return fmt.Errorf("srcContainerName is empty")
+	}
+	if dstContainerName == "" {
+		return fmt.Errorf("dstContainerName is empty")
+	}
+	if acquired := d.volumeLocks.TryAcquire("copy_" + dstContainerName); !acquired {
```

Comment:
where it has `"copy_"` prefix? since `copy_` is a const, there is no difference if you add it or not

Reply:
because in `createVolume` func we also had a volumelock for volumename, I use `copy_` prefix to distinct these two.

Reply:
then don't need to lock here since it's already locked outside

Reply:
done